js/src/forum/components/PostStream.js

```diff
@@ -223,7 +225,7 @@ export default class PostStream extends Component {
       if (time) period = time;
     });
 
-    this.stream.index = index + 1;
+    this.stream.index = indexFromViewPort !== null ? indexFromViewPort + 1 : 1;
```

Comment:
Why remove `$items.first().data('index')`? Although actually, we might be able to solve that other issue in one swoop:

If no posts match, then we are either above or below all posts. Iff we are above all posts, then $items.first().data('index') should be equal to 0. If $items.first().data('index') !== 0, we cannot be above all posts, so we must be below all posts, so we should set index to `this.stream.count()`. I think that makes sense :rofl:?

Reply:
It does make sense. What would be a scenario that we are above all posts? I simplified it (maybe incorrectly) in the latest commit to just assume we are below the posts.

Reply:
On testing, that was an issue with the previous implementation, but under the new "check for null flag" system it doesn't have an issue.